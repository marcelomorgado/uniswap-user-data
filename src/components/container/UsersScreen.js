import React from "react";
import UsersTable from "../presentational/UsersTable";

const data = [
  ["0x123", 159],
  ["0x456", 237],
  ["0x789", 262],
  ["0x987", 305],
  ["0x654", 356],
];

let id = 0;
function createData(userId, etherBalance) {
  id += 1;
  return { id, userId, etherBalance };
}

const rows = [];

for (let i = 0; i < 200; i += 1) {
  const randomSelection = data[Math.floor(Math.random() * data.length)];
  rows.push(createData(...randomSelection));
}

function UsersScreen() {
  return <UsersTable rows={rows} />;
}

export default UsersScreen;
