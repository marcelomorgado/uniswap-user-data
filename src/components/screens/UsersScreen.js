import React, { Fragment } from "react";
import UsersListContainer from "../container/UsersListContainer";
import UserTransactionsModalContainer from "../container/UserTransactionsModalContainer";
import TransferEtherButtonContainer from "../container/TransferEtherButtonContainer";

class UsersScreen extends React.Component {
  state = {
    txModalOpen: false,
    userId: null,
  };

  openTxModal = () => {
    this.setState({ txModalOpen: true });
  };

  closeTxModal = () => {
    this.setState({ txModalOpen: false });
  };

  onRowClick = userId => {
    this.setState({ userId });
    this.openTxModal();
  };

  render() {
    const { state, openTxModal, closeTxModal, onRowClick } = this;
    const { txModalOpen, userId } = state;

    return (
      <Fragment>
        <TransferEtherButtonContainer />
        <UserTransactionsModalContainer
          open={txModalOpen}
          handleOpen={openTxModal}
          handleClose={closeTxModal}
          userId={userId}
        />
        <UsersListContainer onRowClick={onRowClick} />
      </Fragment>
    );
  }
}

export default UsersScreen;
