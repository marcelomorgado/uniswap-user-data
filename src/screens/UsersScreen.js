import React, { Fragment } from "react";
import UsersTableContainer from "../components/container/UsersTableContainer";
import UserTransactionsModalContainer from "../components/container/UserTransactionsModalContainer";

class UsersScreen extends React.Component {
  state = {
    txModalOpen: false,
    userId: null,
  };

  openTxModal = () => {
    this.setState({ txModalOpen: true });
  };

  closeTxModal = () => {
    this.setState({ txModalOpen: false });
  };

  onRowClick = userId => {
    this.setState({ userId });
    this.openTxModal();
  };

  render() {
    const { state, openTxModal, closeTxModal, onRowClick } = this;
    const { txModalOpen, userId } = state;

    return (
      <Fragment>
        <UserTransactionsModalContainer
          open={txModalOpen}
          handleOpen={openTxModal}
          handleClose={closeTxModal}
          userId={userId}
        />
        <UsersTableContainer onRowClick={onRowClick} />
      </Fragment>
    );
  }
}

export default UsersScreen;
