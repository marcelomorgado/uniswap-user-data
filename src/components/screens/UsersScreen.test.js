import React from "react";
import { shallow } from "enzyme";
import UsersScreen from "./UsersScreen";

it("renders without crashing", () => {
  expect(shallow(<UsersScreen />)).toMatchSnapshot();
});
