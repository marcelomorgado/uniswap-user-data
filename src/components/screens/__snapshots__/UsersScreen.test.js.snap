// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Fragment>
  <UserTransactionsModalContainer
    handleClose={[Function]}
    handleOpen={[Function]}
    open={false}
    userId={null}
  />
  <UsersTableContainer
    onRowClick={[Function]}
  />
</Fragment>
`;
